### PR TITLE
feat(cargo-shuttle): beta flag, remove project list pagination logic

### PR DESCRIPTION
**Pull request by jonaro00**
Supersedes #1731

### CODE REVIEW

**Review by Kazy — COMMENTED** [review]
Curious how this will scale if we start to have more modification from the current CLI. I was thinking of having two different implementation of the API instead (`ShuttleApiClient` and `ShuttleBetaApiClient` basically). This would also make it clearer what isn't implemented vs the other one.

**Comment by jonaro00** [comment]
Decided to go with a boolean property like this since the amount of differences thus far is very small.
The pattern becomes:
- If only the API call is different, change behaviour in Client (can also refactor to `<T>` if the return model is different),
- If more of the command is different, change behaviour in the `Shuttle` command method,
- If lots changed, add new methods where needed.

**Comment by Kazy** [comment]
> Decided to go with a boolean property like this since the amount of differences thus far is very small. The pattern becomes:

Okay, let's keep it that way for now then.

File: cargo-shuttle/src/lib.rs [hunk 1]
```diff
@@ -244,9 +240,7 @@ impl Shuttle {
             Command::Project(ProjectCommand::Status { follow }) => {
                 self.project_status(follow).await
             }
-            Command::Project(ProjectCommand::List { page, limit, raw }) => {
-                self.projects_list(page, limit, raw).await
-            }
+            Command::Project(ProjectCommand::List { raw, .. }) => self.projects_list(raw).await,
             Command::Project(ProjectCommand::Stop) => self.project_stop().await,
             Command::Project(ProjectCommand::Delete(ConfirmationArgs { yes })) => {
```
Comment (iulianbarbu) [inline cargo-shuttle/src/lib.rs:245]: Delete should call `DELETE /projects/:project_name` instead of DELETE `/projects/:project_name/delete`.

**Review by iulianbarbu — COMMENTED** [review]
Project start, list, status works as expected.

**Review by iulianbarbu — APPROVED** [review]
Thanks!

**Pull request merged**